RecordCollection/Controllers/AlbumsController.cs

```diff
@@ -15,8 +19,39 @@ public AlbumsController(RecordCollectionContext context)
         public IActionResult Index()
         {
             var albums = _context.Albums.ToList();
-            
             return View(albums);
         }
+
+        [Route("Albums/{id:int}")]
+        public async Task<IActionResult> Show(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var retrieved = await _context.Albums.FindAsync(id);
+
+            var relatedAlbums = _context.Albums
+                .Where(a => a.Artist == retrieved.Artist && a.Id != retrieved.Id)
+                .OrderBy(a => a.ReleaseDate);
+
+            var viewModel = new AlbumViewModel
+            {
+                Album = retrieved,
+                RelatedAlbums = relatedAlbums.ToList()
+            };
+
+            if (viewModel.RelatedAlbums.Count > 0)
```

Comment:
You could also consider taking these if and else statements, and creating a method and calling it here or wherever you may need it. This can make the code more organized and easier to read. This fits one of the refactoring patterns I know. It's called the "Extract Method" pattern.